### PR TITLE
[ticket/17330] Handle mozilla not properly handling webpush padding

### PR DESCRIPTION
**Pull request by marc1706**
PHPBB-17330

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB-17330

### CODE REVIEW

**Review by jrconlin — APPROVED** [review]
The fun of using bridge systems. (Really wish I could get the time to coordinate a proper fix for all of this, but it's not straight forward, at all.)

**Comment by rxu** [comment]
There's no payload error with this fix but also couldn't get any webpush notifications on Firefox + Android subscription so far.

**Comment by rxu** [comment]
Tested and working, thanks.

**Comment by rxu** [comment]
Wondering if it's worth making the setting configurable via ACP or will `2820` value work for everyone.

**Comment by marc1706** [comment]
> Wondering if it's worth making the setting configurable via ACP or will `2820` value work for everyone.

I'd keep it like this for now. Who knows, maybe at some point it'll actually be fixed at Mozilla's end ...

File: phpBB/phpbb/notification/method/webpush.php [hunk 1]
```diff
@@ -442,13 +442,19 @@ protected function clean_expired_subscriptions(array $user_subscription_map, arr
...
 	protected function set_endpoint_padding(\Minishlink\WebPush\WebPush $web_push, string $endpoint): void
 	{
 		if (str_contains($endpoint, 'mozilla.com') || str_contains($endpoint, 'mozaws.net'))
 		{
-			$web_push->setAutomaticPadding(self::MOZILLA_FALLBACK_PADDING);
+			try
+			{
+				$web_push->setAutomaticPadding(self::MOZILLA_FALLBACK_PADDING);
+			}
+			catch (\Exception)
```
[the remark this answers is not in the document]
Comment (marc1706) [reply phpBB/phpbb/notification/method/webpush.php:454]: No, this is not needed anymore in PHP 8.0 and newer if the variable is not used:
https://php.watch/versions/8.0/catch-exception-type

**Pull request merged**